### PR TITLE
libnetwork/networkdb: switch to go-immutable-radix v2

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/buildkit/pull/5055
- relates to https://github.com/moby/moby/pull/48126

**- Description for the changelog**

```markdown changelog

```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: libnetwork/networkdb/networkdb.go [hunk 1]
```diff
@@ -337,15 +337,17 @@ func (nDB *NetworkDB) Peers(nid string) []PeerInfo {
...
 	}
-	if entry != nil && entry.deleting {
+	if v != nil && v.deleting {
 		return nil, types.NotFoundErrorf("entry in table %s network id %s and key %s deleted and pending garbage collection", tname, nid, key)
 	}
-
-	return entry.value, nil
+	if v == nil {
+		return nil, nil
+	}
```
Comment (corhere) [inline libnetwork/networkdb/networkdb.go:349]: Why is this check needed with v2 but not with v1?

Comment (thaJeztah) [reply libnetwork/networkdb/networkdb.go:349]: Good question; maybe it was needed as well; there's a nil-check two lines up, and my IDE was flagging it as "could be nil"; honestly not sure it there's such a chance

Comment (corhere) [reply libnetwork/networkdb/networkdb.go:349]: There is at least one other place which assumes that `(*NetworkDB).getEntry() error = nil` implies that the returned `*entry` is not nil.
https://github.com/moby/moby/blob/c2879978022db54ef21ac93feeabce0cbd75f63d/libnetwork/networkdb/delegate.go#L171-L179
I am pretty sure that it is an invariant. An incomplete audit of the NetworkDB code suggests that to be the case.

**Comment by thaJeztah** [comment]
@corhere updated; I removed the `nil` check, and added a comment there to reference the discussion here.

**Review by corhere — APPROVED** [review]

**Review by vvoland — APPROVED** [review]

**Pull request merged**